rust/simple/main.rs

```diff
@@ -32,20 +32,17 @@ fn main() {
 }
 
 fn try_main() -> Result<(), Box<dyn Error>> {
-    let stdin = io::stdin();
-    let stdin = stdin.lock();
     let mut counts: HashMap<String, u64> = HashMap::new();
-    for result in stdin.lines() {
-        let line = result?;
-        for word in line.split_whitespace() {
-            let canon = word.to_lowercase();
-            *counts.entry(canon).or_insert(0) += 1;
+    for line in io::stdin().lock().lines() {
+        for word in line?.split_whitespace() {
+            *counts.entry(word.to_lowercase()).or_insert(0) += 1;
```

Comment:
I am actually not a fan of `line?`. I purposely used `result` because I think that's clearer.

Squashing `stdin().lock()` down into fewer lines seems OK to me though.

Reply:
Make sense. Both styles are ok for me. I will revert this changes.